benchmarks/nnls/objective.py

```diff
@@ -18,7 +18,7 @@ def set_data(self, X, y):
         self.X, self.y = X, y
 
     def __call__(self, beta):
-        if (beta > 0).all():
+        if (beta < 0).any():
```

Comment:
I don't think it's correct. I would say:

`if (beta >= 0).all():`

Reply:
yes, I was not clear, if you use (beta < 0).any() the action associated is return np.inf